rust/ql/lib/codeql/rust/controlflow/internal/CfgConsistency.qll

```diff
@@ -54,6 +56,24 @@ query predicate deadEnd(CfgImpl::Node node) {
   not letElsePanic(node.getAstNode())
 }
 
+pragma[nomagic]
+private predicate successfullyExtractedFile(File f) {
+  not exists(ExtractionWarning ee | ee.getLocation().getFile() = f)
+}
+
+query predicate missingCfgChild(CfgNode parent, string pred, int i, AstNode child) {
+  CfgNodes::missingCfgChild(parent, pred, i, child) and
+  successfullyExtractedFile(child.getLocation().getFile()) and
+  not exists(AstNode last, CfgImpl::Completion c | CfgImpl::last(child, last, c) |
+    // In for example `if (a && true) ...` there is no RHS CFG node going into the
+    // `[false] a && true` operation
+    strictcount(ConditionalSuccessor cs | exists(last.getACfgNode().getASuccessor(cs)) | cs) = 1
+    or
+    // In for example `x && return`, there is no RHS CFG node going into the `&&` operation
+    not c instanceof CfgImpl::NormalCompletion
```

Comment:
Tweaks the comments a bit. The "no RHS CFG node" makes it sound as if its a node and not an edge that isn't there.
```suggestion
    // In for example `if (a && true) ...`, there is no edge from the CFG node
    // for `true` into the `[false] a && true` node.
    strictcount(ConditionalSuccessor cs | exists(last.getACfgNode().getASuccessor(cs)) | cs) = 1
    or
    // In for example `x && return`, there is no edge from the node for
    // `return` into the `&&` node.
    not c instanceof CfgImpl::NormalCompletion
```